agent/local/state.go

```diff
@@ -1341,22 +1317,17 @@ func (l *State) syncNodeInfo() error {
 	err := l.Delegate.RPC("Catalog.Register", &req, &out)
 	switch {
 	case err == nil:
-		l.nodeInfoInSync = true
 		l.logger.Info("Synced node info")
-		return nil
 
 	case acl.IsErrPermissionDenied(err), acl.IsErrNotFound(err):
 		// todo(fs): mark the node info to be in sync to prevent excessive retrying before next full sync
 		// todo(fs): some backoff strategy might be a better solution
-		l.nodeInfoInSync = true
 		accessorID := l.aclAccessorID(at)
 		l.logger.Warn("Node info update blocked by ACLs", "node", l.config.NodeID, "accessorID", accessorID)
 		metrics.IncrCounter([]string{"acl", "blocked", "node", "registration"}, 1)
-		return nil
 
 	default:
 		l.logger.Warn("Syncing node info failed.", "error", err)
-		return err
```

Comment:
What are the consequences of this function no longer returning an error?

Reply:
Based on your comment I revisited this part and reverted to returning and handling any errors from `syncNodeInfo`. I also added a longer explanation for these changes to the PR description.